goci-interfaces/goci-ui/src/main/java/uk/ac/ebi/spot/goci/ui/controller/GeneImpcLinkController.java

```diff
@@ -0,0 +1,55 @@
+package uk.ac.ebi.spot.goci.ui.controller;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import uk.ac.ebi.spot.goci.ui.component.ImpcIdentifierMap;
+import uk.ac.ebi.spot.goci.ui.constants.SearchUIConstants;
+import uk.ac.ebi.spot.goci.ui.service.GeneImpcLinkService;
+
+import java.util.Optional;
+
+@RestController
+@RequestMapping(value = SearchUIConstants.API_V1 + SearchUIConstants.GENE_URI)
+public class GeneImpcLinkController {
+    private static final Logger log = LoggerFactory.getLogger(GeneImpcLinkController.class);
+
+    @Autowired
+    ImpcIdentifierMap impcIdentifierMap;
+
+    @Autowired
+    GeneImpcLinkService geneImpcLinkService;
+
+    /** API called UI to retreive Mouse Ortholog mapping for Gene
+     *
+     * @param geneId
+     * @return
+     */
+    @ResponseStatus(HttpStatus.OK)
+    @GetMapping(value = "{geneId}/impclink")
+    public String retreiveImpcLink(@PathVariable String geneId ) {
+        String impcId;
+        if (impcIdentifierMap.getImpcGeneMap() != null){
+            impcId = Optional.ofNullable(impcIdentifierMap.getImpcGeneMap()
+                    .get(geneId)).orElse("NA");
+        }
+        else{
+            impcId = "NA";
+        }
+        log.info("Impc id->"+impcId);
+        return impcId;
+    }
+
+    /** API to do on demand load of Mouse Ortholog & Gene Map
+     *
+     */
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @GetMapping(value = "/impclinks")
```

Comment:
There is an http Method "PURGE" that should be used for this purpose (clearing cache), https://stackoverflow.com/questions/25857508/what-is-the-http-method-purge

But its not listed as part of 8 standard methods according to RFC specification, hence spring boot does not allow it. 

I would just use the GetMapping as you have done, but with url: "/v1/impc-mouse-orthologs" with a requestParameter "operation" and value "purge" should trigger the cache purging

/v1/impc-mouse-orthologs?operation=purge